source/common/router/upstream_request.h

```diff
@@ -259,5 +294,29 @@ class HttpUpstream : public GenericUpstream, public Http::StreamCallbacks {
   Http::RequestEncoder* request_encoder_{};
 };
 
+class TcpUpstream : public GenericUpstream, public Tcp::ConnectionPool::UpstreamCallbacks {
```

Comment:
Which timeouts come into play with a CONNECT/WS session? I guess the downstream idle timeouts? Anything else? I think we will need to make sure that https://github.com/envoyproxy/envoy/pull/10531 works with non-pass through CONNECT also? WDYT? cc @Shikugawa 

I haven't reviewed the tests yet but it would be great to make sure we have good integration coverage over all of the various timeout scenarios.

Reply:
All HTTP timeouts may apply - we're 100% reusing the classes which do timeout logic.
Presumably one would want larger (or no) overall timeouts on WS / CONNECT routes

Reply:
I don't think you have to do them all I just wanted to make sure we are covered. @Shikugawa can make sure to cover CONNECT when the linked PR is finished for idle timeouts.